src/service.rs

```diff
@@ -767,6 +811,144 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn always_save_test() -> anyhow::Result<()> {
```

Comment:
This function is quite huge, because I tried to stick to how other tests were written and to contain changes. But I can split it up if you want.

Reply:
It's okay to deviate from the preexisting test structure (especially if it improves their overall quality). The tests are far from a perfect model. :)

Reply:
Then I'll split it up. 😃 Do you want me to extract repeated code used to access headers etc. into separate functions, or perhaps you want to have all unwraps etc. in-place?

Reply:
It would be nice, but don't feel obligated to do more than you want.

Reply:
No problem! 😉